src/main/java/com/github/lordrex34/config/model/ConfigClassInfo.java

```diff
@@ -0,0 +1,160 @@
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.github.lordrex34.config.model;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.lordrex34.config.ConfigManager;
+import com.github.lordrex34.config.annotation.ConfigClass;
+import com.github.lordrex34.config.annotation.ConfigField;
+import com.github.lordrex34.config.postloadhooks.ConfigPostLoadHooks;
+import com.github.lordrex34.config.util.PropertiesParser;
+
+/**
+ * @author NB4L1 (original concept)
+ * @author lord_rex
+ */
+public final class ConfigClassInfo
+{
+	private static final Logger LOGGER = LoggerFactory.getLogger(ConfigClassInfo.class);
+	
+	/** The class that is being scanned for the annotations. */
+	private final Class<?> _clazz;
+	
+	/** The configuration class annotation. */
+	private final ConfigClass _configClass;
+	
+	/** A list of the {@link ConfigFieldInfo} classes contained by this info container class. */
+	private final List<ConfigFieldInfo> _fieldInfoClasses = new ArrayList<>();
+	
+	/**
+	 * Constructs a new info container class.
+	 * @param clazz the configuration class itself
+	 */
+	public ConfigClassInfo(Class<?> clazz)
+	{
+		_clazz = clazz;
+		_configClass = Objects.requireNonNull(_clazz.getDeclaredAnnotation(ConfigClass.class), () -> "Class " + _clazz + " doesn't have @ConfigClass annotation!");
+		
+		for (Field field : _clazz.getDeclaredFields())
+		{
+			if (field.getDeclaredAnnotation(ConfigField.class) == null)
+			{
+				// skip fields not using ConfigField annotation
+				continue;
+			}
+			
+			_fieldInfoClasses.add(new ConfigFieldInfo(_clazz, field));
+		}
+	}
+	
+	/**
+	 * Loads the configuration class that is being managed by this information container.
+	 * @param overriddenProperties please see {@link ConfigManager#getOverriddenProperties}
+	 */
+	public void load(PropertiesParser overriddenProperties)
+	{
+		final Path configPath = Paths.get("", _configClass.pathNames()).resolve(_configClass.fileName() + _configClass.fileExtension());
+		if (Files.notExists(configPath))
+		{
+			LOGGER.warn("Config File {} doesn't exist! Generating ...", configPath);
+			
+			try
+			{
+				generate();
+			}
+			catch (IOException e)
+			{
+				LOGGER.warn("Failed to generate config!", e);
+			}
```

Comment:
( won't it fail the shit out of the loading few lines below if the file is missing? why swallow it here if it stops it later anyway? :D )

Reply:
Done.